mlflow/tracking/fluent.py

```diff
@@ -1308,20 +1309,27 @@ def print_auto_logged_info(r):
     }
 
     def setup_autologging(module):
-        autolog_fn = LIBRARY_TO_AUTOLOG_FN[module.__name__]
         try:
-            needed_params = list(inspect.signature(autolog_fn).parameters.keys())
-            filtered = {k: v for k, v in locals_copy if k in needed_params}
-        except ValueError:
-            filtered = {}
+            autolog_fn = LIBRARY_TO_AUTOLOG_FN[module.__name__]
+            try:
+                needed_params = list(inspect.signature(autolog_fn).parameters.keys())
+                filtered = {k: v for k, v in locals_copy if k in needed_params}
+            except ValueError:
+                filtered = {}
 
-        try:
             autolog_fn(**filtered)
             _logger.info("Autologging successfully enabled for %s.", module.__name__)
         except Exception as e:  # pylint: disable=broad-except
-            _logger.warning(
-                "Exception raised while enabling autologging for %s: %s", module.__name__, str(e)
-            )
+            if _is_testing():
```

Comment:
Allows `mlflow.autolog()` to respect test mode, enabling us to catch errors raised by `mlflow.autolog()`.